packages/babel-plugin-proposal-decorators/src/transformer-2022-03.ts

```diff
@@ -937,7 +937,12 @@ function transformClass(
         class extends ${state.addHelper("identity")} {}
       ` as t.ClassExpression;
       staticsClass.body.body = [
-        t.staticBlock([t.toStatement(path.node, false)]),
+        t.staticBlock([
+          t.toStatement(originalClass, true) ||
+            // If toStatement returns false, originalClass must be an anonymous ClassExpression,
+            // because `export default @dec ...` has been handled in the export visitor before.
+            t.expressionStatement(originalClass as t.ClassExpression),
+        ]),
```

Comment:
Would it be better to fix in `t.toStatement`?

Also unrelated to this PR, the type definition of `t.toStatement` doesn't seem quite right.

Reply:
> Would it be better to fix in t.toStatement?

I don't know much context here but it seems that we intentionally fail `toStatement` for anonymous class expression

https://github.com/babel/babel/blob/53185ea275b0a278ac4aafe5841f2a332cabff31/packages/babel-types/test/converters.js#L131-L138

/cc @nicolo-ribaudo if you can shred some light here.

If we decided to overturn this behaviour, we still need the patch here until Babel 8 because the plugin does not depend on `@babel/types`.

Reply:
https://github.com/liuxingbaoyu/babel/commit/98ca541fde0c1ec9f2b5e5222e0adf08740cb540

Good find, now I think it's fine.

Reply:
I believe the reason is that `t.toStatement` of a `ClassExpression` should return a `ClassDeclaration`, but converting a class expression without an ID is generally not safe.